packages/core/src/components/spinner/spinner.tsx

```diff
@@ -159,13 +99,56 @@ export class Spinner extends AbstractPureComponent<SpinnerProps> {
             return SpinnerSize.STANDARD;
         }
         return Math.max(MIN_SIZE, size);
-    }
+    }, [className, size]);
+
+    // keep spinner track width consistent at all sizes (down to about 10px).
+    const strokeWidth = Math.min(MIN_STROKE_WIDTH, (STROKE_WIDTH * SpinnerSize.LARGE) / sizePx);
+    const strokeOffset = PATH_LENGTH - PATH_LENGTH * (value == null ? 0.25 : clamp(value, 0, 1));
 
     /** Compute viewbox such that stroked track sits exactly at edge of image frame. */
-    private getViewBox(strokeWidth: number) {
+    const viewBox = React.useMemo(() => {
         const radius = R + strokeWidth / 2;
         const viewBoxX = (50 - radius).toFixed(2);
         const viewBoxWidth = (radius * 2).toFixed(2);
         return `${viewBoxX} ${viewBoxX} ${viewBoxWidth} ${viewBoxWidth}`;
-    }
-}
+    }, [strokeWidth]);
+
+    const classes = classNames(
+        Classes.SPINNER,
+        Classes.intentClass(intent),
+        { [Classes.SPINNER_NO_SPIN]: value != null },
+        className,
+    );
+
+    // multiple DOM elements around SVG are necessary to properly isolate animation:
+    // - SVG elements in IE do not support anim/trans so they must be set on a parent HTML element.
```

Comment:
> // - SVG elements in IE do not support anim/trans so they must be set on a parent HTML element.

Not sure if this is still comment is still relevant since we haven't supported IE11 for a while now. Regardless, I have tested these changes to Spinner in Edge.